03-SQL_Advanced/exercises.md

```diff
@@ -23,11 +51,40 @@ Here is how the output looks like (list of countries is truncated)
  
 You will need to know this function in PostgreSQL: [string_agg()](https://www.dbrnd.com/2016/09/postgresql-string_agg-to-concatenate-string-per-each-group-like-sql-server-stuff-string-aggregation-function/). This is an aggregate function that allows you to list out all strings.
 
+```SQL
+SELECT region_name, (y2016::float/y1985) as growth,
+      (CASE WHEN (y2016::float/y1985)>=3.652 THEN 'High Growth' ELSE 'Low Growth' END) as comparision_growth
+FROM (SELECT region_name, 
+      sum(CASE WHEN year=1985 THEN gdp_per_capita ELSE NULL END) AS y1985,
+      sum(CASE WHEN year=2016 THEN gdp_per_capita ELSE NULL END) AS y2016
+      FROM public.gdp
+      GROUP BY region_name) as original
+WHERE y2016 IS NOT NULL AND y1985 IS NOT NULL
+```
+**UGH HOW DO YOU DO THIS PROBLEM!!!!!!**
+
```

Comment:
Not sure if I am correct, but based on the proposed output, the ```region_name``` needs to be aggregated using the string_agg() function and not out on its own. And then we can group by the growth (what you have as 'comparison_growth'. 

Based on your code it would look something like this: 

```SQL
SELECT  
      (CASE WHEN (y2016::float/y1985)>=3.652 THEN 'High Growth' ELSE 'Low Growth' END) as comparison_growth,
      STRING_AGG(region_name,', ') 
FROM (SELECT region_name, 
      sum(CASE WHEN year=1985 THEN gdp_per_capita ELSE NULL END) AS y1985,
      sum(CASE WHEN year=2016 THEN gdp_per_capita ELSE NULL END) AS y2016
      FROM public.gdp
      GROUP BY region_name) as original
WHERE y2016 IS NOT NULL AND y1985 IS NOT NULL
GROUP BY comparison_growth;
```